### ISSUE TITLE
Add support for enum with flags

### ISSUE THREAD

**Issue by AmenJlili** [SoftCircuits/WinSettings#1]
I'm not too sure WinSettings currently supports enum with flags. Any plans to support it @SoftCircuits ?

**Comment by SoftCircuits** [comment]
@AmenJlili

> I'm not too sure WinSettings currently supports enum with flags. Any plans to support it @SoftCircuits ?

Sorry, I have no idea what you're referring to. Maybe you can show an example that doesn't work as you want it to.

**Comment by AmenJlili** [comment]
I'm getting exception in my code :

>System.Exception: 'Settings property 'CopyToTheseDocuments' is an unsupported data type 'XXXXXXX.CopyRevision.DocumentType'. Change property type or use ExcludedSetting attribute.'

CopyToTheseDocuments is an enum with flags.

@SoftCircuits

**Comment by SoftCircuits** [comment]
I see what you're saying. I don't believe flags is the issue. It's just an unknown type and the code doesn't know what to do with it.

I think the easiest fix is to convert it to an `int`.

    public int DocumentTypeInt { get; set; }

    [ExcludeSetting]
    public DocumentType DocumentType
    {
        get => (DocumentType)DocumentTypeInt;
        set => DocumentTypeInt = (int)value;
    }

**Comment by AmenJlili** [comment]
Do you think you can add support so that winsettings can recognize enums?

**Comment by SoftCircuits** [comment]
I can look into this. I'm not sure when I'll get to it though.

**Comment by AmenJlili** [comment]
@SoftCircuits : I will create a PR myself if I get to this on my end!

**Comment by SoftCircuits** [comment]
> @SoftCircuits : I will create a PR myself if I get to this on my end!

@AmenJlili I expect to have a little time this weekend.

**Comment by SoftCircuits** [comment]
@AmenJlili So, I looked into this a bit. The WinSettings library supports encrypting settings, and this is done through my EasyEncryption library. Because of this, the WinSettings library supports the same types that the EasyEncryption library supports. I spent a little while looking at this. Adding support for enums to EasyEncryption is not trivial. I will think about it some more over the weekend. Right now, I'm not sure if this is something I will do. You are welcome to take a look if you want.
